### PR TITLE
expression: fix invalid compare-operations in vectorized `builtinInDecimalSig `

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
When comparing two decimals, we should use `MyDecimal.Compare` instead of using `==`.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_other_vec_generated.go [hunk 1]
```diff
@@ -188,7 +188,7 @@ func (b *builtinInDecimalSig) vecEvalInt(input *chunk.Chunk, result *chunk.Colum
 			arg0 := args0[i]
 			arg1 := args1[i]
 			compareResult = 1
```
Comment (lonng) [inline expression/builtin_other_vec_generated.go]: Seems we don't need this variable?

Comment (qw4990) [reply expression/builtin_other_vec_generated.go]: This logic is generated from our template and this variable makes it easier for us to write template code.

Comment (lonng) [reply expression/builtin_other_vec_generated.go]: Got

**Comment by breeswish** [comment]
LGTM

**Pull request merged**